File: src/01-easy/ascii-art/solution.js
const L=+readline(),H=+readline(),T=readline().toUpperCase();
const rows=[];for(let i=0;i<H;i++)rows.push(readline());
for(let i=0;i<H;i++){
    let line='';
    for(const c of T){let idx=c.charCodeAt(0)-65;if(idx<0||idx>25)idx=26;line+=rows[i].substr(idx*L,L);}
    console.log(line);
}

File: src/01-easy/ascii-art/solution.scala
object Solution extends App {
  val L=scala.io.StdIn.readInt();val H=scala.io.StdIn.readInt()
  val T=scala.io.StdIn.readLine().toUpperCase
  val rows=Array.fill(H)(scala.io.StdIn.readLine())
  for(i<-0 until H){
    println(T.map{c=>val idx={val i=c-'A';if(i<0||i>25)26 else i};rows(i).substring(idx*L,idx*L+L)}.mkString)
  }
}
